frontend/src/ui/Access.tsx

```diff
@@ -525,44 +627,31 @@ const Warning: React.FC<WarningProps> = ({ tooltip, info }) => (
     </WithTooltip>
 );
 
-const UnchangeableAllActions = () => {
+const UnchangeableAllActions: React.FC<{ permission?: PermissionLevel }> = ({ permission }) => {
     const { t } = useTranslation();
-    return <span css={{ marginLeft: 8 }}>{t("manage.access.table.actions.write")}</span>;
+    const { permissionLevels } = useAclContext();
+    const label = permission ?? permissionLevels.highest;
+    return <span css={{ marginLeft: 8 }}>{t(`manage.access.table.actions.${label}`)}</span>;
 };
 
 const ActionsMenu: React.FC<ItemProps> = ({ item, kind }) => {
     const isDark = useColorScheme().scheme === "dark";
     const ref = useRef<FloatingHandle>(null);
     const { t } = useTranslation();
-    const { change } = useAclContext();
-    const currentActionOption = item.actions.has("write") ? "write" : "read";
-    const changeOption = (newOption: "read" | "write") => change(prev => {
-        notNullish(prev.get(item.role)).actions = new Set(
-            newOption === "write" ? ["read", "write"] : ["read"]
-        );
-    });
-
-
-    const actions: Action[] = ["read", "write"];
-
-    const count = kind === "User" ? 1 : 2;
-    const translations = (actionType: Action) => match(actionType, {
-        "read": () => ({
-            label: t("manage.access.table.actions.read"),
-            description: t("manage.access.table.actions.read-description", { ...{ count } }),
-        }),
-        "write": () => ({
-            label: t("manage.access.table.actions.write"),
-            description: t("manage.access.table.actions.write-description", { ...{ count } }),
-        }),
-    });
+    const { change, permissionLevels } = useAclContext();
+    const allLabels = Object.keys(permissionLevels.all) as PermissionLevel[];
+    const currentActionOption = getActionLabel(item, permissionLevels);
 
+    const changeOption = (newOption: PermissionLevel) => change(
+        prev => notNullish(prev.get(item.role)).actions
+            = notNullish(permissionLevels.all[newOption]).actions
+    );
```

Comment:
```suggestion
    const changeOption = (newOption: PermissionLevel) => change(prev => {
        notNullish(prev.get(item.role)).actions 
            = notNullish(permissionLevels.all[newOption]).actions;
    });
```

I suppose thats closer to the style we use elsewhere.